components/application-operator/pkg/utils/string_map.go

```diff
@@ -0,0 +1,45 @@
+package utils
+
+const (
+	Separator     string = "."
+	RuneSeparator        = '.'
+)
+
+type InterfaceMap map[string]interface{}
+type StringMap map[string]string
+
+func NewStringMap(source InterfaceMap) StringMap {
+	releaseMap := make(StringMap)
+	releaseMap.flatten("", source)
+	return releaseMap
+}
+
+func (m StringMap) ContainsAll(labels StringMap) bool {
+	if len(m) != len(labels) {
+		return false
+	}
+
+	for key, applicationValue := range labels {
+		if releaseVal, ok := m[key]; !ok || applicationValue != releaseVal {
+			return false
+		}
+	}
+
+	return true
+}
+
+func (m StringMap) flatten(parentKey string, source map[string]interface{}) {
```

Comment:
It could be a bit shorter:
```
func flatten(prefix string, source InterfaceMap) {	
	for key, value := range source {
		newKey := prefix + key
		
		if casted, ok := value.(InterfaceMap); ok {
			flatten(newKey + Separator, casted)
		} else {
			m[newKey] = value.(string)
		}
	}
}
```

Flatten method could be also implemented as a regular function (not a method receiver):

```
func flatten(prefix string, source InterfaceMap, target StringMap) {	
	for key, value := range source {
		newKey := prefix + key
		
		if casted, ok := value.(InterfaceMap); ok {
			flatten(newKey + Separator, casted, target)
		} else {
			target[newKey] = value.(string)
		}
	}
}
```
NewStringMap could be implemented as follows:
```
func NewStringMap(source InterfaceMap) StringMap {
	return flatten(source, make(StringMap))
}
```

Reply:
Corrected according to the first snippet. Do you have in mind any performance benefits in terms of using functions over methods?